internal/commands/cnab.go

```diff
@@ -76,13 +80,27 @@ func duffleHome() home.Home {
 }
 
 // prepareDriver prepares a driver per the user's request.
-func prepareDriver(dockerCli command.Cli) (driver.Driver, error) {
+func prepareDriver(dockerCli command.Cli, bindLocalSocket bool) (driver.Driver, error) {
 	driverImpl, err := driver.Lookup("docker")
 	if err != nil {
 		return driverImpl, err
 	}
 	if d, ok := driverImpl.(*driver.DockerDriver); ok {
 		d.SetDockerCli(dockerCli)
+		if bindLocalSocket {
+			d.AddConfigurationOptions(func(config *container.Config, hostConfig *container.HostConfig) error {
+				config.User = "0:0"
+				mounts := []mount.Mount{
+					{
+						Type:   mount.TypeBind,
+						Source: "/var/run/docker.sock",
```

Comment:
Shouldn't the source be retrieved from the `targetContext`, for the same reason that @ijc said https://github.com/docker/app/pull/478/files#r264752546 ?